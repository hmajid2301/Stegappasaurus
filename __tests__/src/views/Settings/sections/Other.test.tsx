import { mount, shallow } from "enzyme";
import React from "react";
import configureStore, { MockStoreEnhanced } from "redux-mock-store";

import OtherConnected, { Other } from "~/views/Settings/sections/Other";

describe("Other: Match Snapshots", () => {
  test("1", () => {
    const component = shallow(
      <Other color="#17212D" background="#FFF" resetPreferences={() => null} />
    );
    expect(component).toMatchSnapshot();
  });

  test("2", () => {
    const component = shallow(
      <Other color="#FFF" background="#17212D" resetPreferences={() => null} />
    );
    expect(component).toMatchSnapshot();
  });
});

describe("Other: Props", () => {
  test("OnPress", async () => {
    const component = mount(
      <Other color="#FFF" background="#17212D" resetPreferences={() => null} />
    );
    const prop = component
      .find("TouchableOpacity")
      .last()
      .props() as any;
    prop.onPress();
    expect(prop).toBeTruthy();
  });
});

describe("Other: Redux", () => {
  let store: MockStoreEnhanced<unknown, {}>;
  beforeAll(() => {
    const mockStore = configureStore();
    const initialState = {};
    store = mockStore(initialState);
  });

  test("mapDispatchToProps", () => {
    const component = shallow(
      //@ts-ignore
      <OtherConnected color="#FFF" background="#17212D" store={store} />
    );
    (component.props() as any).resetPreferences();
    const actions = store.getActions();
    expect(actions).toMatchSnapshot();
  });
});
