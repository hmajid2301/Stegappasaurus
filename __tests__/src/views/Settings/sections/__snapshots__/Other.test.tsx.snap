// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Other: Match Snapshots 1 1`] = `
<View>
  <Themed.ListItem
    containerStyle={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    title={
      <Text
        style={
          Object {
            "color": "#009CFF",
            "fontFamily": "Roboto",
            "fontSize": 12,
          }
        }
      >
        Other
      </Text>
    }
    titleStyle={
      Object {
        "paddingBottom": 5,
        "paddingTop": 20,
      }
    }
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    title={
      <View>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Roboto-Light",
                },
                Object {
                  "color": "#17212D",
                },
              ]
            }
          >
            Reset Preferences
          </Text>
        </TouchableOpacity>
      </View>
    }
    topDivider={true}
  />
</View>
`;

exports[`Other: Match Snapshots 2 1`] = `
<View>
  <Themed.ListItem
    containerStyle={
      Object {
        "backgroundColor": "#17212D",
      }
    }
    title={
      <Text
        style={
          Object {
            "color": "#009CFF",
            "fontFamily": "Roboto",
            "fontSize": 12,
          }
        }
      >
        Other
      </Text>
    }
    titleStyle={
      Object {
        "paddingBottom": 5,
        "paddingTop": 20,
      }
    }
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#17212D",
      }
    }
    title={
      <View>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Roboto-Light",
                },
                Object {
                  "color": "#FFF",
                },
              ]
            }
          >
            Reset Preferences
          </Text>
        </TouchableOpacity>
      </View>
    }
    topDivider={true}
  />
</View>
`;

exports[`Other: Redux mapDispatchToProps 1`] = `
Array [
  Object {
    "payload": Object {},
    "type": "RESET_PREFERENCES",
  },
]
`;
